### PR TITLE
[cgal] update to 5.5.1

### PR DESCRIPTION
**Pull request by lrineau**
**Describe the pull request**

- #### What does your PR fix?
  Update CGAL to version 5.5.1

- #### Which triplets are supported/not supported? Have you updated the [CI baseline](https://github.com/microsoft/vcpkg/blob/master/scripts/ci.baseline.txt)?
  all

- #### Does your PR follow the [maintainer guide](https://github.com/microsoft/vcpkg/blob/master/docs/maintainers/maintainer-guide.md)?
  yes

- #### If you have added/updated a port: Have you run `./vcpkg x-add-version --all` and committed the result?
  yes

### CODE REVIEW

File: ports/cgal/vcpkg.json [hunk 1]
```diff
@@ -1,6 +1,6 @@
 {
   "name": "cgal",
-  "version": "5.5",
+  "version": "5.5.1",
   "port-version": 1,
```
Comment (MonicaLiu0311) [inline ports/cgal/vcpkg.json:4]:
```suggestion
```

Comment (MonicaLiu0311) [reply ports/cgal/vcpkg.json:4]: Whenever `version` changes, `port-version` must be set to 0, so please remove it.

Comment (lrineau) [reply ports/cgal/vcpkg.json]: 🤦 done with last push! Thanks.

**Comment by MonicaLiu0311** [comment]
Feature `qt` is tested successfully in the following triplet:
x86-windows
x64-windows
x64-windows-static

**Review by MonicaLiu0311 — APPROVED** [review]

**Pull request merged**